src/main/java/online/book/store/controller/BookController.java

```diff
@@ -28,30 +29,35 @@ public class BookController {
 
     private final BookService bookService;
 
+    @PreAuthorize("hasRole('ROLE_USER')")
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of all available books")
     public List<BookDto> findAll(Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
+    @PreAuthorize("hasRole('ROLE_USER')")
     @GetMapping("/{id}")
     @Operation(summary = "Get book by ID", description = "Get the existing book by ID")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBook(id);
     }
```

Comment:
Users with `admin` role should have access to `getBookById()` method also

Reply:
If an individual user needs additional access, we can grant them an additional role.
For example, super admin can have two roles in his parameter Set<Role> roles(ADMIN and USER) instead of hardcoding roles for each endpoint